packages/rainbowkit/src/wallets/walletConnectors/enkryptWallet/enkryptWallet.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { ConnectorNotFoundError, Ethereum } from '@wagmi/core';
+import { getAddress } from 'ethers/lib/utils';
+import { InjectedConnector } from 'wagmi/connectors/injected';
+import { Chain } from '../../../components/RainbowKitProvider/RainbowKitChainContext';
+import { Wallet } from '../../Wallet';
+
+export interface EnkryptWalletOptions {
+  chains: Chain[];
+  shimDisconnect?: boolean;
+}
+
+interface EIP5749ConnectorOptions {
+  walletId: string;
+  chains: Chain[];
+  options: {
+    shimDisconnect: boolean | undefined;
+  };
+}
+const detectProvider = (
+  walletId: string
+): {
+  isAvailable: boolean;
+  provider?: Ethereum;
+} => {
+  const isAvailable =
+    typeof window !== 'undefined' &&
+    typeof (window as any).evmproviders !== 'undefined' &&
+    (window as any).evmproviders[walletId]
+      ? true
+      : false;
+  if (isAvailable)
+    return {
+      isAvailable,
+      provider: (window as any).evmproviders[walletId],
+    };
+  return { isAvailable };
+};
+class EIP5749Connector extends InjectedConnector {
+  walletId: string;
+  ready: boolean;
+  constructor(options: EIP5749ConnectorOptions) {
+    const injectorOptions = {
+      getProvider: () =>
+        typeof window !== 'undefined' &&
+        typeof (window as any).evmproviders !== 'undefined'
+          ? (window as any).evmproviders[options.walletId]
+          : undefined,
+      ...options.options,
+    };
+    super({
+      chains: options.chains,
+      options: injectorOptions,
+    });
+    this.walletId = options.walletId;
+    this.ready = detectProvider(this.walletId).isAvailable;
+    setTimeout(() => {
+      this.ready = detectProvider(this.walletId).isAvailable;
+    }, 1000); //Chrome MV3 inject immediate is still not 100% functioning therefor sometimes there is a delay
+  }
+
+  async getAccount() {
+    const provider = await super.getProvider();
+    if (!provider) throw new ConnectorNotFoundError();
+    return provider
+      .request({
+        method: 'eth_accounts',
+      })
+      .then((accounts: string[]) => getAddress(accounts[0] as string))
+      .catch((err: unknown) => err); //somewhere in import { createClient} from 'wagmi', it doesnt handle the eth_accounts error properly thats why it keeps popping up with connect window. This fixes it;
+  }
+}
+
+export const enkryptWallet = ({
+  chains,
+  shimDisconnect,
+}: EnkryptWalletOptions): Wallet => {
+  return {
+    createConnector: () => {
+      return {
+        connector: new EIP5749Connector({
+          chains,
+          options: {
+            shimDisconnect,
+          },
+          walletId: 'enkrypt',
+        }),
+      };
+    },
+    downloadUrls: {
+      browserExtension: 'https://www.enkrypt.com',
+    },
+    iconBackground: '#fff',
+    iconUrl: async () => (await import('./enkryptWallet.svg')).default,
+    id: 'enkrypt',
+    installed: true,
```

Comment:
`installed` is used incorrectly; RainbowKit uses this value to determine when to show the user the connect state versus the "Get Wallet" state. Should check if wallet is injected and provide a boolean or undefined.

Reply:
this is required due to the fact that, with Manifest V3 injected wallets currently it wont inject before the page load, so when the library check whether an injected wallet exists it'll return null. Thats why you have to check it after a few millisecond delay and set installed accordingly

Reply:
@kvhnuke RainbowKit uses `installed` for UI gating, including displaying a fallback injected wallet for unsupported wallets. We do actually need that property to be accurate. Every Manifest v3 wallet that bumped into this issue did find a workaround: https://github.com/rainbow-me/rainbowkit/pull/1042#issuecomment-1505331023

Reply:
unfortunately those are workarounds and there is chrome bug conversation happening
finally they made the fix on Chrome 111 but current chrome version is 112, so We are waiting for atleast Chrome version 116 to release the fix
https://bugs.chromium.org/p/chromium/issues/detail?id=634381

Reply:
anyway Ill push a simplified code that doesnt touch installed

Reply:
> @kvhnuke RainbowKit uses `installed` for UI gating, including displaying a fallback injected wallet for unsupported wallets. We do actually need that property to be accurate. Every Manifest v3 wallet that bumped into this issue did find a workaround: [#1042 (comment)](https://github.com/rainbow-me/rainbowkit/pull/1042#issuecomment-1505331023)

also xdefi wallet is Manifest v2

Reply:
Going through similar issue in the manifest v3  -  #1041